src/components/Dropdowns/SelectionDropdown.tsx

```diff
@@ -217,6 +219,8 @@ export default function SelectionDropdown(inputProps: SelectionDropdownProps): R
         isDisabled={props.disabled}
         isClearable={false}
         isSearchable={props.isSearchable}
+        // TODO: Allow `onChange` to be async, and show a loading indicator
+        // + the awaited value while waiting for it to resolve.
```

Comment:
This is a change I've wanted to make for a while, and it's already written up as #390